src/awkward/_connect/numpy.py

```diff
@@ -14,75 +14,64 @@
 
 
 def convert_to_array(layout, args, kwargs):
-    out = ak._v2.operations.to_numpy(layout, allow_missing=False)
+    out = ak.operations.to_numpy(layout, allow_missing=False)
     if args == () and kwargs == {}:
         return out
     else:
         return numpy.array(out, *args, **kwargs)
 
 
-# implemented = {}
+implemented = {}
 
 
-# def _to_rectilinear(arg):
-#     if isinstance(
-#         arg,
-#         (
-#             ak.Array,
-#             ak.Record,
-#             ak.ArrayBuilder,
-#             ak._v2.contents.Content,
-#             ak._v2.record.Record,
-#             ak.layout.ArrayBuilder,
-#             ak.layout.LayoutBuilder32,
-#             ak.layout.LayoutBuilder64,
-#         ),
-#     ):
-#         nplike = ak.nplike.of(arg)
-#         return nplike.to_rectilinear(arg)
-#     else:
-#         return arg
-
-
-# def array_function(func, types, args, kwargs):
-#     function = implemented.get(func)
-#     if function is None:
-#         args = tuple(_to_rectilinear(x) for x in args)
-#         kwargs = dict((k, _to_rectilinear(v)) for k, v in kwargs.items())
-#         out = func(*args, **kwargs)
-#         nplike = ak.nplike.of(out)
-#         if isinstance(out, nplike.ndarray) and len(out.shape) != 0:
-#             return ak.Array(out)
-#         else:
-#             return out
-#     else:
-#         return function(*args, **kwargs)
+def _to_rectilinear(arg):
+    if isinstance(arg, tuple):
+        nplike = ak.nplike.of(*arg)
+        return tuple(nplike.to_rectilinear(x) for x in arg)
+    else:
+        nplike = ak.nplike.of(arg)
+        nplike.to_rectilinear(arg)
+
+
+def array_function(func, types, args, kwargs):
```

Comment:
Nice! You enabled NEP 18. Are there any v2 tests that test it?

Reply:
The test suite hits the `array_function` implementation in three test-files (although I can't say how extensively).